tests/phpunit/tests/http/http.php

```diff
@@ -66,6 +69,12 @@ public function make_absolute_url_testcases() {
 
 			// Schemeless URL's (not valid in HTTP Headers, but may be used elsewhere).
 			array( '//example.com/sub/', 'https://example.net', 'https://example.com/sub/' ),
+
+			// URLs with fragments.
+			array( '/path#frag', 'http://example.org/', 'http://example.org/path#frag' ),
+			array( '/path/#frag', 'http://example.org/', 'http://example.org/path/#frag' ),
+			array( '/path#frag&ment=1', 'http://example.org/', 'http://example.org/path#frag&ment=1' ),
+			array( '/path?query=string#frag', 'http://example.org/', 'http://example.org/path?query=string#frag' ),
```

Comment:
A test containing `%23` (URL encoded hash) would be lovely.

Reply:
Added in [5a18500](https://github.com/WordPress/wordpress-develop/pull/2983/commits/5a18500cd0c7f636b33c87e4e09e6d577021f225).